second-analysis-steps/code/18-switch-mass-hypo/ntuple_switchHypo.py

```diff
@@ -15,18 +15,19 @@
 # configure an algorithm to substitute the Kaon in the D0-decay by a second
 # pion
 subs = SubstitutePID(
-    'MakeD02pipi',
-    Code="DECTREE('[D*(2010)+ -> (D0 -> K- pi+) pi+]CC')",
+    'MakeD02Kpi',
+    Code="DECTREE('[D*(2010)+ -> (D0 -> K- K+) pi+]CC')",
     # note that SubstitutePID can't handle automatic CC
     Substitutions={
-        'Charm -> (D0 -> ^K- pi+) Meson': 'pi-',
-        'Charm -> (D~0 -> ^K+ pi-) Meson': 'pi+'
-    }
+        'Charm -> (D0 -> ^K- K+) Meson': 'pi-',
```

Comment:
Won't this change `D⁰ → K⁻ K⁺` to `D⁰ → π⁻ π⁺` instead of `D⁰ → K⁻ π⁺`? I think it should be:
```python
Substitutions={
    'Charm -> (D0 -> K- ^K+) Meson': 'pi+',
    'Charm -> (D~0 -> K+ ^K-) Meson': 'pi-'
}
```

Reply:
This also applies to #98.

Alternatively, everything could be done with `D⁰ → π⁻ π⁺` as the alternative mass hypothesis.

Reply:
That will need to be checked, but for the time being I'm ok with doing just these substitutions.

Reply:
Yeah, let's keep it simple.